single-line-diagram-viewer/src/main/java/com/powsybl/sld/viewer/SingleLineDiagramViewer.java

```diff
@@ -279,81 +273,62 @@ private ContainerDiagramResult createContainerDiagramView(Container c) {
-            } catch (IOException e) {
-                throw new UncheckedIOException(e);
-            }
-            return new ContainerDiagramResult(diagramView, svgData, metadataData, jsonData);
+            loadSvg(diagramView, svgData);
+
+            return new ContainerDiagramResult(svgData, metadataData, jsonData);
         }
 
-        private void handleSwitchPositionchange(Container c, String switchId) {
-            Switch sw = null;
-            if (c.getContainerType() == ContainerType.VOLTAGE_LEVEL) {
-                VoltageLevel v = (VoltageLevel) c;
-                sw = v.getNetwork().getSwitch(switchId);
-            } else if (c.getContainerType() == ContainerType.SUBSTATION) {
-                Substation s = (Substation) c;
-                sw = s.getNetwork().getSwitch(switchId);
-            }
-            if (sw != null) {
-                sw.setOpen(!sw.isOpen());
-                DiagramStyleProvider styleProvider = styles.get(styleComboBox.getSelectionModel().getSelectedItem());
-                styleProvider.reset();
-                loadDiagram(c);
+        protected void loadSvg(WebView diagramView, String svgInputStream) {
+            // convert svg file to WebView components
+            try {
+                String b = "<html>" +
+                        String.format("<body><div id='%1$s'>", divId) +
+                        svgInputStream +
+                        "</div></body></html>";
+
+                diagramView.getEngine().loadContent(b);
+
+                // install node and wire handlers to allow diagram edition
+                // FIXME: add handlers with js
+            } catch (Exception e) {
+                // to feed the content of the 'SVG' and 'Metadata' tab, even if the
+                // svg diagram cannot be loaded by svg loader
+                LOGGER.warn("Error in loading svg image : {}", e.getMessage());
             }
         }
 
         private void loadDiagram(Container c) {
-            Service<ContainerDiagramResult> loader = new Service<ContainerDiagramResult>() {
-                @Override
-                protected Task<ContainerDiagramResult> createTask() {
-                    return new Task<ContainerDiagramResult>() {
-                        @Override
-                        protected ContainerDiagramResult call() {
-                            return createContainerDiagramView(c);
-                        }
-                    };
+            WebView diagramView = new WebView();
+            diagramView.setZoom(1.0);
+            ContainerDiagramResult result = createContainerDiagramView(diagramView, c);
+            flowPane.setContent(diagramView);
+            // Adjust size to content
+            diagramView.prefHeightProperty().bind(flowPane.heightProperty());
+            diagramView.prefWidthProperty().bind(flowPane.widthProperty());
+
+            // Add Zoom management
+            diagramView.addEventFilter(ScrollEvent.SCROLL, (ScrollEvent e) -> {
```

Comment:
Catching scroll events makes it difficult to use the scrollbars, could you test if ctrl key is pressed on with `isControlDown()`? Like it usually is in browsers.